generator/src/test/java/com/scottlogic/deg/generator/cucumber/engine/utils/CucumberTestHelper.java

```diff
@@ -7,53 +7,76 @@
 import com.scottlogic.deg.generator.Field;
 import com.scottlogic.deg.generator.GenerateExecute;
 import com.scottlogic.deg.generator.guice.BaseModule;
+import com.scottlogic.deg.generator.inputs.InvalidProfileException;
 import org.junit.Assert;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * Responsible for generating data in cucumber tests.
  */
 public class CucumberTestHelper {
-
     private final CucumberTestState testState;
 
     public CucumberTestHelper(CucumberTestState testState){
         this.testState = testState;
     }
 
-    public List <List<Object>> generateAndGetData() {
-        if (testState.dataGenerationType == null) {
-            throw new RuntimeException("Gherkin error: Please specify the data strategy");
-        }
-
-        if (this.generatorHasRun()) {
-            return testState.generatedObjects;
-        }
+    public void runGenerationProcess() {
+        if (testState.generationHasAlreadyOccured) return;
 
         try {
             Module concatenatedModule =
                 Modules
-                .override(new BaseModule(new CucumberGenerationConfigSource(testState)))
-                .with(new CucumberTestModule(testState));
+                    .override(new BaseModule(new CucumberGenerationConfigSource(testState)))
+                    .with(new CucumberTestModule(testState));
 
             Injector injector = Guice.createInjector(concatenatedModule);
 
             injector.getInstance(GenerateExecute.class).run();
-
-            return testState.generatedObjects;
         } catch (Exception e) {
             testState.addException(e);
-            return null;
         }
+
+        testState.generationHasAlreadyOccured = true;
+    }
+
+    public List<List<Object>> generateAndGetData() {
+        if (testState.shouldSkipGeneration()) {
+            throw new RuntimeException(
+                "Gherkin error: Don't use profile validity steps in conjunction with data checking steps");
+        }
+
+        if (testState.dataGenerationType == null) {
+            throw new RuntimeException("Gherkin error: Please specify the data strategy");
+        }
+
+        if (this.generatorHasRun()) {
+            return testState.generatedObjects == null
+                ? new ArrayList<>()
+                : testState.generatedObjects;
+        }
+
+        runGenerationProcess();
+
+        return testState.generatedObjects == null
+            ? new ArrayList<>()
+            : testState.generatedObjects;
+    }
+
+    public void runPreGenerationChecks() {
```

Comment:
This implies that generation can be run afterwards, but the state would be left in a 'disabled' state?

How about a different name, e.g. `checkProfile` or `runGeneratorForProfileChecks`?